docs/research/ha/RATIONALE.md

```diff
@@ -0,0 +1,446 @@
+# High Availability with s3gw
+
+- [High Availability with s3gw](#high-availability-with-s3gw)
+  - [Active-Active pipelines](#active-active-pipelines)
+  - [Active-Passive pipelines](#active-passive-pipelines)
+  - [One Active pipeline](#one-active-pipeline)
+  - [Investigation's Rationale](#investigations-rationale)
+  - [Failure cases](#failure-cases)
+    - [radosgw's POD failure](#radosgws-pod-failure)
+    - [Cluster's node failure](#clusters-node-failure)
+    - [radosgw's failure due to a bug not related to a certain input pattern](#radosgws-failure-due-to-a-bug-not-related-to-a-certain-input-pattern)
+    - [radosgw's failure due to a bug related to a certain input pattern](#radosgws-failure-due-to-a-bug-related-to-a-certain-input-pattern)
+    - [PV Data corruption at application level due to radosgw's bug](#pv-data-corruption-at-application-level-due-to-radosgws-bug)
+    - [PV Data corruption at application level due to radosgw's anomalous exit](#pv-data-corruption-at-application-level-due-to-radosgws-anomalous-exit)
+  - [Measuring s3gw failures on Kubernetes](#measuring-s3gw-failures-on-kubernetes)
+  - [The s3gw Probe](#the-s3gw-probe)
+  - [Tested scenarios](#tested-scenarios)
+    - [EXIT-1, 10 measures](#exit-1-10-measures)
+    - [EXIT-0, 10 measures](#exit-0-10-measures)
+    - [Considerations on the tests performed](#considerations-on-the-tests-performed)
+
+We want to investigate what *High Availability* - HA - means for a project like
+the s3gw.
+
+If we identify the meaning of HA as the ability to have N independent pipelines
+so that the failure of some of these does not affect the user's operations,
+this is something that could be not easy to achieve with the s3gw.
+
+With pipeline we mean all the chain from the ingress to the `PV`:
+
+```mermaid
+flowchart LR
+    ingress(ingress) ==> radosgw(radosgw) ==> PV((PV))
+```
+
+The difficulty with a project like the s3gw is due to the fact that basically
```

Comment:
Actually, this is also an advantage, due to its lower complexity. Active/active syncing of all operations is non-trivial, and an active/standby pattern is much simpler to implement with lower overhead in the absence of faults.